src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,38 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """
+    Add target information to run_info under target_data
+    :param string scope: The running scope
+    :param string target: The target that we want to store info for
+    :param string or list of items to make into a key key: The key for the info being stored
+    :param dict or string val: The value of the info being stored
+    """
+    # Stringify list of keys to make one key
+    if not isinstance(key, basestring):
+      new_key = [str(key_item) for key_item in key]
+      key = '#'.join(new_key)
+
+    if target in self._target_data:
+      if scope in self._target_data[target]:
+        self._target_data[target][scope].update({key: val})
+      else:
+        self._target_data[target].update({scope: {key: val}})
+    else:
+      self._target_data.update({target: {scope: {key: val}}})
+
+  def report_test_info(self, scope, target, key, test_info):
```

Comment:
I don't think `RunTracker` should gain awareness of tests... given that this method is called from `junit` and `pytest`, it seems like maybe `pants.task.testrunner_task_mixin.TestRunnerTaskMixin` would be a better place for it?